src/alchemlyb/tests/parsing/test_amber.py

```diff
@@ -23,15 +23,20 @@ def invalid_file(request):
                          [filename
                           for leg in load_simplesolvated()['data'].values()
                           for filename in leg])
-def test_dHdl(filename,
-              names=('time', 'lambdas'),
-              shape=(500, 1)):
+def test_dHdl(filename, names=('time', 'lambdas'), shape=(500, 1)):
     """Test that dHdl has the correct form when extracted from files."""
     dHdl = extract_dHdl(filename, T=300)
-
     assert dHdl.index.names == names
     assert dHdl.shape == shape
 
+@pytest.mark.parametrize("filename",
```

Comment:
For this PR, one test is enough, we don't want to increase the CI time with parametrize(.

Reply:
This was my reasoning for which I tested for this fix in the old test. Testing for correct time-reading within the old test makes sense because the old test just checked if the dHdl/u_nk reader worked, checking for two parameters, I just added the time parameter to be reviewed, without increasing the time spent in "parameterize(". If you want I can revert to the old version which didn't slow down anything!

Reply:
So the incentive of the unit test is to make the debugging easier. When I change a piece of code, did it break the code? Is it like the code not running at all? Or it runs but didn't give the correct structure? Or it doesn't gives the correct numerical value?

The test `test_dHdl` tests if the function will run and it will not give out invalid results, so it covers all files but only check the basic structure of the output.

The test `test_dHdl_time_reading` will test if the correct time will be read.

So if `test_dHdl` passes, but `test_dHdl_time_reading` didn't, we know that the codes runs and the framework has been setup correctly but the time handling has been wrong.

If both of them failed, we know that the code didn't run at all.

Merging the two tests would make it harder to debug.

My suggestion is that instead of read all the files and check the time for each file, just run one file and check if the time is correct.